File: app/src/main/java/starbright/com/projectegg/features/detail/RecipeDetailContract.java
package starbright.com.projectegg.features.detail;

import java.util.List;

import starbright.com.projectegg.data.local.model.Ingredient;
import starbright.com.projectegg.data.local.model.Instruction;
import starbright.com.projectegg.features.base.BasePresenter;
import starbright.com.projectegg.features.base.BaseView;

class RecipeDetailContract {

    interface View extends BaseView<RecipeDetailPresenter> {
        void showProgressBar();
        void hideProgressBar();

        void hideScrollContainer();
        void showScrollContainer();
        void hideEmptyStateTextView();

        void renderErrorStateTextView(String errorMessage);
        void renderEmptyStateTextView();
        void renderBannerFoodImage(String imageURL);

        void renderHeaderContainer(int serving, int preparationMinutes, int cookingMinutes, String recipeName);
        void renderIngredientCard(List<Ingredient> ingredients);
        void renderInstructionCard(List<Instruction> instructions);

        void setupSwipeRefreshLayout();

        void setupAdView();

        void createShareIntent(String url, String recipeName);

        void navigateToWebViewActivity(String url);
    }

    interface Presenter extends BasePresenter {
        void setRecipeId(String recipeId);
        void getRecipeDetailInformation(String recipeId);

        void handleShareMenuClicked();

        void handleWebViewMenuClicked();
    }
}

File: app/src/main/java/starbright/com/projectegg/features/detail/RecipeDetailContract.kt
/*
 * Copyright (c) by Andreas (oentoro.andreas@gmail.com)
 * created at 31 - 7 - 2020.
 */

/**
 * Created by Andreas on 5/10/2019.
 */

/**
 * Created by Andreas on 22/9/2018.
 */

package starbright.com.projectegg.features.detail

import androidx.annotation.StringRes
import starbright.com.projectegg.data.model.Ingredient
import starbright.com.projectegg.data.model.Instruction
import starbright.com.projectegg.features.base.BasePresenterContract
import starbright.com.projectegg.features.base.BaseViewContract

class RecipeDetailContract {

    interface View : BaseViewContract {
        fun showProgressBar()
        fun hideProgressBar()

        fun hideScrollContainer()
        fun showScrollContainer()
        fun hideEmptyView()

        fun renderErrorView(errorMessage: String)
        fun renderEmptyView()
        fun renderBannerFoodImage(imageURL: String)

        fun renderHeaderContainer(
            serving: Int,
            cookingMinutes: Int,
            recipeName: String,
            dishType: String,
            calories: Int
        )

        fun renderIngredientsList(ingredients: MutableList<Ingredient>)
        fun renderInstructionsList(instructions: MutableList<Instruction>)

        fun setupSwipeRefreshLayout()

        fun createShareIntent(url: String, recipeName: String)

        fun navigateToWebViewActivity(url: String)

        fun showSnackbar(@StringRes text: Int)

        fun updateMenu(isBookmarked: Boolean)
    }

    interface Presenter : BasePresenterContract {
        fun getRecipeDetailInformation(recipeId: String)
        fun handleShareMenuClicked()
        fun handleWebViewMenuClicked()
        fun handleBookmarkRecipeMenuClicked()
    }
}